### ISSUE TITLE
(Resolved) run stuck

### ISSUE THREAD

**Issue by leiyangly** [WeichenZhou/PALMER#25]
Hi Arthur, I'm reporting an issue that causes PALMER to get stuck before loading the genome index. By reading PALMER.v2.0.0.cpp, it looks like that it was caused by the program not able to properly take ARGV variables when ran as a shortcut. Running PALMER from the original executable file resolved this issue.

**Comment by WeichenZhou** [comment]
Thank you for the posting, Lei.

Please try not to use the relative pathway to run PALMER executable file. I will troubleshoot this for the next upgrade.

Best,
Arthur

**Comment by clifflun** [comment]
This problem still exists in v2.0.1

PALMER cannot run by adding the binary to the env $PATH string

**Comment by WeichenZhou** [comment]
Thank you for using palmer. Can you post what you did/commands?

**Comment by clifflun** [comment]
Thank you for writing this program

I tried to add the PALMER binary to env $PATH string and run the program

It gets stuck after printing "ref is CRCh38"

This problem is resolved when I use the full absolute path of the binary, I am guessing this is the same problem that the original poster had

> Thank you for using palmer. Can you post what you did/commands?

**Comment by WeichenZhou** [comment]
Right. PALMER would still have trouble on the non-absolute path in some cases. Could you please post your command lines that have errors or got stuck so that I can get troubleshooting? 

Thank you!

> Thank you for writing this program
> 
> I tried to add the PALMER binary to env $PATH string and run the program
> 
> It gets stuck after printing "ref is GRCh38"
> 
> https://github.com/WeichenZhou/PALMER/blob/4942234d183f1469f3b421ec5c7209414b817fa9/PALMER.v2.0.0.cpp#L337
> 
> This problem is resolved when I use the full absolute path of the binary, I am guessing this is the same problem that the original poster had
> 
> > Thank you for using palmer. Can you post what you did/commands?

**Comment by clifflun** [comment]
![image](https://github.com/user-attachments/assets/af0043bf-c8e4-4c26-a50c-493b0108f194)
![image](https://github.com/user-attachments/assets/64372a79-2da3-4031-8f37-e5b49cdbd2f5)
There are no errors, it is just stuck

**Comment by WeichenZhou** [comment]
ok so this is the version before got resolved? for both command lines and $PATH?

**Comment by clifflun** [comment]
hmm not entirely sure now if I am using 2.0.0 or 2.0.1, but I installed PALMER following the steps in the documentation

`git clone https://github.com/mills-lab/PALMER.git
cd PALMER
make`
